daemon/daemon.go

```diff
@@ -610,13 +622,18 @@ func (daemon *Daemon) waitForNetworks(c *container.Container) {
 			if _, ok := err.(libnetwork.ErrNoSuchNetwork); !ok {
 				continue
 			}
-			// use a longish timeout here due to some slowdowns in libnetwork if the k/v store is on anything other than --net=host
-			// FIXME: why is this slow???
+
+			// Restart the timer that was allocated outside of this loop
+			timer.Reset(dur)
+
 			logrus.Debugf("Container %s waiting for network to be ready", c.Name)
 			select {
 			case <-daemon.discoveryWatcher.ReadyCh():
-			case <-time.After(60 * time.Second):
+			case <-timer.C:
 			}
+
+			// Don't run the timer while we don't need it.
+			timer.Stop()
```

Comment:
You already have `defer timer.Stop()` above, I guess it's not needed there if you have a stop here.

Reply:
`timer.Stop` here ensures that we aren't running the timer in the loop when we don't need to. The defer ensures the timer is stopped on return.

Reply:
Oh I see what you mean, yeah, no need here.